interfaces/cython/cantera/test/test_thermo.py

```diff
@@ -422,6 +422,125 @@ def test_equil_results(gas, fuel, ox, Y_Cf, Y_Of, Y_Co, Y_Oo, basis):
 
         test_equil_results(gas, fuel, ox, Y_Cf, Y_Of, Y_Co, Y_Oo, 'mass')
 
+    def test_equivalence_ratio_simple_dilution(self):
+        gas = ct.Solution("gri30.yaml")
+
+        gas.X = "H2:4,O2:1,CO:3,CO2:4,N2:5,CH4:6"
+        state = gas.state
+        self.assertNear(gas.equivalence_ratio(include_species=["H2", "O2"]), 2)
+        self.assertNear(gas.equivalence_ratio("H2", "O2", include_species=["H2", "O2"]), 2)
+        self.assertArrayNear(state, gas.state)
+
+        def test_simple_dilution(fraction,basis):
+            if isinstance(fraction, str):
+                fraction_type = fraction[:fraction.find(":")]
+            else:
+                fraction_type = list(fraction.keys())[0]
+
+            M_H2 = gas.molecular_weights[gas.species_index("H2")]
+            M_O2 = gas.molecular_weights[gas.species_index("O2")]
+
+            gas.set_equivalence_ratio(2, "H2", "O2", fraction=fraction, diluent="CO2", basis=basis)
+            if basis == "mole" and fraction_type == "diluent":
+                self.assertNear(gas["H2"].X[0], (1 - 0.3) * 0.8)
+                self.assertNear(gas["O2"].X[0], (1 - 0.3) * 0.2)
+                self.assertNear(gas["CO2"].X[0], 0.3)
+            elif basis == "mass" and fraction_type == "diluent":
+                self.assertNear(gas["H2"].Y[0] / gas["O2"].Y[0], 4 * M_H2 / M_O2)
+                self.assertNear(gas["CO2"].Y[0], 0.3)
+            elif basis == "mole" and fraction_type == "fuel":
+                self.assertNear(gas["H2"].X[0], 0.1)
+                self.assertNear(gas["O2"].X[0], 0.1 / 4)
+                self.assertNear(gas["CO2"].X[0], 1 - 0.1 - 0.1 / 4)
+            elif basis == "mass" and fraction_type == "fuel":
+                self.assertNear(gas["H2"].Y[0], 0.1)
+                self.assertNear(gas["H2"].Y[0] / gas["O2"].Y[0], 4 * M_H2 / M_O2)
+            elif basis == "mole" and fraction_type == "oxidizer":
+                self.assertNear(gas["H2"].X[0], 0.1 * 4)
+                self.assertNear(gas["O2"].X[0], 0.1)
+                self.assertNear(gas["CO2"].X[0], 1 - 0.1 - 0.1 * 4)
+            elif basis == "mass" and fraction_type == "oxidizer":
+                self.assertNear(gas["O2"].Y[0], 0.1)
+                self.assertNear(gas["H2"].Y[0] / gas["O2"].Y[0], 4 * M_H2 / M_O2)
+
+            state = gas.state
+            self.assertNear(gas.equivalence_ratio("H2", "O2", include_species=["H2", "O2"], basis=basis), 2)
+            self.assertNear(gas.equivalence_ratio(include_species=["H2", "O2"], basis=basis), 2)
+            self.assertArrayNear(state, gas.state)
+
+        # brute force all possible input combinations
+        test_simple_dilution("diluent:0.3",   "mole")
+        test_simple_dilution({"diluent":0.3}, "mole")
```

Comment:
```suggestion
        test_simple_dilution("diluent:0.3", "mole")
        test_simple_dilution({"diluent": 0.3}, "mole")
```
(here and elsewhere)